pkg/kubelet/volumes.go

```diff
@@ -200,7 +200,9 @@ func (kl *Kubelet) getPodVolumes(podUID types.UID) ([]*volumeTuple, error) {
 			if volumeNameDir != nil {
 				volumes = append(volumes, &volumeTuple{Kind: volumeKind, Name: volumeNameDir.Name()})
 			} else {
-				glog.Errorf("Could not read directory %s: %v", podVolDir, volumeNameDirsStat[i])
+				// failed syscall type error, expose it
+				glog.Errorf("Failure: Could not read directory %s: %v", podVolDir, volumeNameDirsStat[i])
+				return volumes, fmt.Errorf("Failure: Could not read directory: \n  Path: %s \n  Error: %v \n", volumeKindPath, volumeNameDirsStat[i])
```

Comment:
remove "Failure"

error messages must start with a lowercase letter

remove the newlines ("error reading directory %q: %v", volumeKindPath, volumeNameDirsStat[i])